### PR TITLE
Script the creation of the arm manifest files

### LINKED ISSUE #423: Proposal generate the static yaml manifests from the helm chart
## Context

We could simplify the maintenance and ensure the consistency of the Helm chart and the static yaml manifests by using the `helm template` command to generate the static files.  This will require some additional modifications to remove the helm specific labels/annotations.

Generating the manifests will ensure that they stay in sync with any changes in the Helm chart.

### PR DESCRIPTION
**Pull request by LucasRoesler**
## Description

- Tweak create-static-manifest to allow values files not in the chart
directory
- Add values files to set the docker images for the arm64 and armhf
- Update make to generate the arm manifest when we package the helm
chart

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Resolves #423 

## How Has This Been Tested?

I have not been able to directly test this because I do not have any arm based clusters on hand

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
On ARMHF:

```
 kubectl logs deploy/alertmanager -n openfaas
addrs 0.0.0.0:9094 
level=info ts=2019-04-26T13:08:30.752057377Z caller=main.go:145 msg="Starting Alertmanager" version="(version=0.15.0-rc.0, branch=HEAD, revision=68122e7005958df611baec01b316355a6f408bae)"
level=info ts=2019-04-26T13:08:30.75229688Z caller=main.go:146 build_context="(go=go1.10, user=root@46262c45c129, date=20180228-12:10:52)"
level=info ts=2019-04-26T13:08:30.818442364Z caller=main.go:271 msg="Loading configuration file" file=/alertmanager.yml
level=error ts=2019-04-26T13:08:30.826994235Z caller=main.go:274 msg="Loading configuration file failed" file=/alertmanager.yml err="unknown fields in basic_auth: password_file"
```

This means we should:

* template auth off for yaml_armhf
* and update the contrib Dockerfile for AlertManager and Prometheus to match the ones used by the helm charts

**Comment by LucasRoesler** [comment]
@alexellis  What about arm64?

Also, i noticed some typos in the faas project for the new armhf version, the make scripts don't tag with the matching version .  Meaning, this is the make cmds in the alertmanager 0.16.1 folder
```

.PHONY: ci-armhf-build
ci-armhf-build:
	docker build -t functions/alertmanager:0.15.0-armhf .

.PHONY: ci-armhf-push
ci-armhf-push:
	docker push functions/alertmanager:0.15.0-armhf
```

I will create a PR in faas

**Comment by LucasRoesler** [comment]
@alexellis i pushed typo fixes in the faas repo https://github.com/openfaas/faas/pull/1173

**Comment by alexellis** [comment]
I think we need a rebase on this now, can you fix?

I'll add the ARM64 Dockerfiles to contrib folder and push new versions if you can also update values*.yml

**Comment by alexellis** [comment]
Basic auth should be off for armhf and arm64 as not to break existing tutorials.

**Comment by alexellis** [comment]
Docker images have been pushed for these new versions on ARM64 and ARMHF:

https://github.com/openfaas/faas/commit/f511feb140ae92f71ba1b5a0cca1037f6eecf0bc

**Comment by alexellis** [comment]
Doing more testing on this. FYI the "functions/alertmanager:v0.16.1-arm64" should be formatted without a "v" so just "functions/alertmanager:0.16.1-arm64" and so-forth.

**Comment by alexellis** [comment]
More testing shows that the functions are being installed in the openfaas namespace, I'd suggest adding the namespace default into the values file for arm*

**Comment by LucasRoesler** [comment]
@alexellis i have addressed your comments and then I squashed the commits

**Comment by LucasRoesler** [comment]
The testing error seems to be an issue with kustomize

```
48.02s$ ./contrib/get_kind.sh
go get -u sigs.k8s.io/kind
# sigs.k8s.io/kind/pkg/kustomize
../../../sigs.k8s.io/kind/pkg/kustomize/kustomize.go:108:7: undefined: k8sdeps.NewFactory
../../../sigs.k8s.io/kind/pkg/kustomize/kustomize.go:109:26: not enough arguments in call to build.NewCmdBuild
The command "./contrib/get_kind.sh" exited with 2.
```

**Comment by alexellis** [comment]
Had this with ofc-bootstrap too, they've switched to go modules

**Comment by LucasRoesler** [comment]
Should i copy this then? https://github.com/openfaas-incubator/ofc-bootstrap/commit/50410e016ebb33d81ab82bcdb673a901b465b948

**Comment by alexellis** [comment]
Since we build in docker it should be a no harm change.

**Comment by LucasRoesler** [comment]
I guess i need to be more specific with enabling go modules

```
go: finding golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
go: finding github.com/gogo/protobuf v1.1.1
go get: error loading module requirements

The command "./contrib/get_kind.sh" exited with 1.
0.01s$ ./contrib/create_kind_cluster.sh
./contrib/create_kind_cluster.sh: line 3: kind: command not found
The command "./contrib/create_kind_cluster.sh" exited with 127.
```

**Comment by alexellis** [comment]
@LucasRoesler the way I did this for OFC-bootstrap was to enable it only where the go get statement was used, this means it does not affect anything else built in the environment.

Could be simpler to just add: 
```
export GO111MODULE="on"
```

Before the `go get` - it also decouples it from `travis`.

Do you want to try it that way?

Alex

**Comment by LucasRoesler** [comment]
@alexellis finally fixed the CI scripts

File: contrib/get_kind.sh [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 echo "go get -u sigs.k8s.io/kind"
 
-go get -u sigs.k8s.io/kind
+export GO111MODULE="on"
+go get -u sigs.k8s.io/kind@master
```
Comment (alexellis) [inline contrib/get_kind.sh]: Given the comments we had from the maintainers do you think that is a good idea to use HEAD/master? Is there a more suitable version that we can pin to or not?

Comment (LucasRoesler) [reply contrib/get_kind.sh]: I just updated it with the current master sha, since we know that works right now: kubernetes-sigs/kind@0a6ceae

**Comment by alexellis** [comment]
This requires some more on-device testing before merging.

**Comment by LucasRoesler** [comment]
@alexellis i have rebased on mater and regenerated the files.  This includes the new basic auth plugin now

**Comment by alexellis** [comment]
Great, thank you. The way this helped with helm/x86 was awesome. So much easier to maintain one set of files.

This change will now need regression testing on armhf and arm64 before merging. We also need to consider what tutorials people are following for armhf for OpenFaaS and whether this is a breaking change as configured right now.

**Comment by alexellis** [comment]
Thank you so much for this PR, it's going to save a lot of time :tada: 

I was unable to merge the conflicts made to binary files. I would recommend against committing binaries in future PRs (as a general rule)

**Comment by alexellis** [comment]
Merged via #458

**Pull request closed without merging**